core/trie/trie.go

```diff
@@ -752,6 +753,7 @@ func (t *Trie) Iterate(startValue *felt.Felt, consumer func(key, value *felt.Fel
 
 func (t *Trie) doIterate(startKey, key *Key, consumer func(key, value *felt.Felt) (bool, error)) (bool, error) {
 	if key == nil {
+		// shouldn't it be finished == true here?
 		return false, nil
```

Comment:
This is suspicious - should it be `finished == true` here?

Reply:
Returning false is finish right?